test_requests.py

```diff
@@ -411,6 +416,21 @@ def test_DIGESTAUTH_QUOTES_QOP_VALUE(self):
         r = requests.get(url, auth=auth)
         assert '"auth"' in r.request.headers['Authorization']
 
+    def test_DIGESTAUTH_THREADED(self):
```

Comment:
I was indeed agressive, just to make sure. I can't think another way beyond the agressive bruteforce.

testing race conditions is always tricky, will probably require white box testing, and synchronisation test hooks inside the production code. I am not so found of these technics